### PR TITLE
feat:  Adds in ParseOption classes

### PR DESCRIPTION
**Pull request by Zemill**
This class along with its associated code allows the users to set different options for parsing.  Some default options are already defined (minimal, typical, and full)

The purpose of this change is to reduce cycle time of the parser parsing data that is not used by the end user.  Things like mouse events and messages are removed from the Typical ParseOptions option.

Some minor syntax changes.  These  changes were originally written by  @poma and @martijnhoekstra and consolidated by @zemill

### CODE REVIEW

**Comment by barrett777** [comment]
Thanks for putting this together!  Could you resolve the merge conflicts?

**Comment by Zemill** [comment]
Ok. should be good to go now

File: Heroes.ReplayParser/ParseOptions.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public class ParseOptions
+    {
+        public bool IgnoreErrors { get; set; } = false;
+        public bool AllowPTR { get; set; } = false;
+        public bool ShouldParseEvents { get; set; } = true;
+        public bool ShouldParseUnits { get; set; } = false;
+        public bool ShouldParseMouseEvents { get; set; } = false;
+        public bool ShouldParseDetailedBattleLobby { get; set; } = true;
+        public bool ShouldParseMessageEvents { get; set; } = false;
+        public bool ShouldParseStatistics { get; set; } = true;
```
Comment (barrett777) [inline Heroes.ReplayParser/ParseOptions.cs:27]: Looks good overall :) Yes I've needed an options class like this for awhile

One request: Could we set the defaults to match what the current defaults are, to hopefully avoid some confusion.  Users can of course specify whatever they want

The change would be to enable 'ShouldParseUnits', enable 'ShouldParseMouseEvents', and disable 'ShouldParseDetailedBattleLobby'

**Review by barrett777 — CHANGES_REQUESTED** [review]

File: Heroes.ReplayParser/ParseOptions.cs (hunk 1 above)
Comment (Zemill) [reply Heroes.ReplayParser/ParseOptions.cs:27]: I have committed some changes to work with that.  I have removed TypicalParsing and replaced it with DefaultParsing.  I then created MediumParsing to capture what was previous TypicalParsing.  I also set the default parsing mode to DefaultParsing.

Take a look and let me know if that looks right

**Review by barrett777 — APPROVED** [review]

**Pull request merged**